tests/Elastic.CommonSchema.Benchmarks/SerializingStringBuilderBase.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Text;
+using AutoBogus;
+using BenchmarkDotNet.Attributes;
+
+namespace Elastic.CommonSchema.Benchmarks
+{
+	[UnicodeConsoleLogger, MemoryDiagnoser, ThreadingDiagnoser]
+	public class SerializingStringBuilderBase
+	{
+		[Benchmark]
+		public StringBuilder Empty()
+		{
+			var ecs = new Base();
+			return ecs.Serialize(new StringBuilder());
```

Comment:
Curious about these benchmarks, since the whole idea is to test the performance of reusing the same StringBuilder, instead of allocating new strings. Would think the StringBuilder should be a member-variable initialized outside the benchmark-code.

Reply:
In the NLog integration we get a `StringBuilder` from `RenderFormattedMessage` which as far as I can tell is not guaranteed to be passed a cached/reused `StringBuilder` either.

Reply:
NLog Targets makes use of an object-pool of StringBuilders. So yes NLog will cache/reuse the same StringBuilder again and again. Which is why it is recommended to write "directly" to StringBuilder instead of allocating strings.

Reply:
I think there should be some additional benchmarks to also compare with a reused `StringBuilder`, as that will measure reuse in conjunction with reusing json writer on the same thread.

Reply:
Happy to amend, I went of the worst case scenario:

https://github.com/NLog/NLog/blob/29879ece25a7d2e47a148fc3736ec310aee29465/src/NLog/Layouts/Layout.cs#L278

Where `RenderAllocateBuilder` is called within Nlog without a `reusableBuilder`. We don't have the same infrastructure NLog has with https://github.com/NLog/NLog/blob/86fba05e272a07d508b32475957b9310f607061d/src/NLog/Internal/ObjectPools/StringBuilderPool.cs

The benchmark is not indicative of how fast NLog can go, but rather tests the worst case scenario where someone passes a new StringBuilder to our `Serialize()` method which to me still yields acceptable performance.

The reuse of json writer is not depending on passing the same StringBuilder either.